### PR TITLE
Upgrade to enableIf 1.1.8

### PR DESCRIPTION
**Pull request by da-tubi**
Big thanks to @Atry for reviewing https://github.com/ThoughtWorksInc/enableIf.scala/pull/70 in a patient and kind way!

This PR simplifies enableIf using the handy function `classpathMatches`.

### CODE REVIEW

File: src/main/scala/ai/eto/rikai/sql/spark/RikaiSparkSessionExtensions.scala [hunk 1]
```diff
@@ -73,21 +74,13 @@ private class MlPredictRule(val session: SparkSession)
     }
   }
 
-  @enableIf(c =>
-    c.classPath.exists(
-      _.getPath.matches(".*spark-catalyst_2\\.\\d+-3\\.2\\..*")
-    )
-  )
+  @enableIf(classpathMatches(".*spark-catalyst_2\\.\\d+-3\\.2\\..*".r))
```
Comment (eddyxu) [inline src/main/scala/ai/eto/rikai/sql/spark/RikaiSparkSessionExtensions.scala:77]: Can it check against "not spark 3.1"?

This enableIf does not work with 3.3+ right?  3.3 is being voted at the moment.

Comment (darcy-shen) [reply src/main/scala/ai/eto/rikai/sql/spark/RikaiSparkSessionExtensions.scala]: the latest regex check against "not spark 3.0 or 3.1", please review again

Comment (da-tubi) [reply src/main/scala/ai/eto/rikai/sql/spark/RikaiSparkSessionExtensions.scala]: The latest regex check against "not spark 3.1 or 3.0", please review again

File: build.sbt [hunk 2]
```diff
@@ -73,7 +73,7 @@ libraryDependencies ++= {
   val scalatestVersion = "3.2.0"
   val circeVersion = "0.12.3"
   val mlflowVersion = "1.21.0"
-  val enableifVersion = "1.1.7"
+  val enableifVersion = "1.1.7+82-56fd8fb0"
```
Comment (da-tubi) [inline build.sbt:76]: https://github.com/ThoughtWorksInc/enableIf.scala/releases/tag/v1.2.0 will be available tmr

Comment (eddyxu) [reply build.sbt:76]: Shall we wait then?

**Review by eddyxu — APPROVED** [review]
LGTM overall. I'd suggest we wait for a `enableif` new version to be released.

File: build.sbt (hunk 2 above)
Comment (da-tubi) [reply build.sbt:76]: Yes, I will wait.

**Pull request merged**